code-challenges.test.js

```diff
@@ -15,37 +15,228 @@
 // a) Create a test with an expect statement using the variable provided.
 // HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+// const colors1 = ["purple", "blue", "green", "yellow", "pink"]
 // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = [
-  "chartreuse",
-  "indigo",
-  "periwinkle",
-  "ochre",
-  "aquamarine",
-  "saffron"
-]
+// const colors2 = [
+//   "chartreuse",
+//   "indigo",
+//   "periwinkle",
+//   "ochre",
+//   "aquamarine",
+//   "saffron"
+// ]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
+// pseudocode : (1:25 pm) 03.10.2023 I want to create an function.
+// what does this function do ? : it takes in an array as an argument, removes the first item from the array and shuffles the remaining content.
+// what will I call this function ? : colorShuffler 
+// how do I make this function take in an array, remove the first item from the array and shuffle the remaining content ?  I can use the The arrayContaining matcher ?(1:34pm) 
+describe('colorShuffler', () => {
+  const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+  const colors2 = [
+    "chartreuse",
+    "indigo",
+    "periwinkle",
+    "ochre",
+    "aquamarine",
+    "saffron"
+  ]
+  ;
+  it('it takes in an array, removes the first item from the array and shuffles the remaining content', () => {
+    // invoking my function colorShuffler passing in the value of colors1 an array containing colors. 
+    expect(colorShuffler(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"])); 
+    expect(colorShuffler(colors2)).toEqual (expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+  });
+});
+
+// test out put:
+// ReferenceError: colorShuffler is not defined
+
 // b) Create the function that makes the test pass.
+const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+const colors2 = ["chartreuse", "indigo",  "periwinkle", "ochre", "aquamarine","saffron" ]
 
+const colorShuffler = (arr) => {
+  // (2:00pm) Start from the last element and swap
+  // one by one. I don't need to run for
+  // the first element that's why i > 0
+  for (let i = arr.length - 1; i > 0; i--) {
```

Comment:
Awesome job, but we still need to first remove the first item of the given array. So this should have an output that can be in any order but cannot have "purple" for colors1 and "chartreuse" for colors2